pippy/PipelineStage.py

```diff
@@ -663,9 +666,13 @@ def forward_maybe_with_nosync(self, *args, **kwargs):
             out_val = self.submod(*args, **kwargs)
         return out_val
 
-    def backward_maybe_with_nosync(self, bwd_kwargs: Dict, is_last_chunk: bool):
+    def backward_maybe_with_nosync(self, bwd_kwargs: Dict, bwd_chunk_id: int):
+        # Add `stage_info` for debug purpose in `stage_backward`
+        bwd_kwargs["stage_info"] = f"{self.stage_index}"
```

Comment:
how is the debug information used?

Reply:
Just like this, pretty dumb:
```
    except Exception as e:
        exc_msg = f"""
        Failed to run backward on stage {stage_info}
        Stage output: {map_debug_info(stage_output)}
        Output gradient: {map_debug_info(output_grads)}
        Input: {map_debug_info(input_values)}
        """
        raise RuntimeError(exc_msg) from e
```
I could remove this argument later.

Reply:
Removed :)